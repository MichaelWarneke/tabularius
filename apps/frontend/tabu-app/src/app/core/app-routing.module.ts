import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
