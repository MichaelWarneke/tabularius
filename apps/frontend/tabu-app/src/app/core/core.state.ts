import {
  ActionReducerMap,
  MetaReducer,
  createFeatureSelector
} from '@ngrx/store';
import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { RouterStateUrl } from '@tabularius/shared/services';
import { storeFreeze } from 'ngrx-store-freeze';

import { initStateFromLocalStorageReducer } from '@tabularius/shared/services';
import { debugReducer } from '@tabularius/shared/services';

import { environment } from '../../environments/environment';

export interface AppState {
  router: RouterReducerState<RouterStateUrl>;
}

export const reducers: ActionReducerMap<AppState> = {
  router: routerReducer
};

export const metaReducers: MetaReducer<AppState>[] = [
  initStateFromLocalStorageReducer
];
if (!environment.production) {
  metaReducers.unshift(storeFreeze);
  metaReducers.unshift(debugReducer);
}

export const selectRouterState = createFeatureSelector<
  AppState,
  RouterReducerState<RouterStateUrl>
>('router');
