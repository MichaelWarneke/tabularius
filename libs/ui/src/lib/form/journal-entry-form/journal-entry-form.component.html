<h1>Journal Form</h1>
<tabu-dynamic-form
  [formModel]="formModel"
  [data]="journal"
  (saveEntry)="onSave($event)"
></tabu-dynamic-form>
