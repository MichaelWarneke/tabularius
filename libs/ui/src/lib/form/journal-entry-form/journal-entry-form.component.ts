import {
  Component,
  OnInit,
  ChangeDetectionStrategy,
  Output,
  EventEmitter,
  Input
} from '@angular/core';
import { IJournal } from '@tabularius/shared/models';
import { Validators, FormBuilder } from '@angular/forms';

export class JournalModel {
  date = [{ value: '', disabled: true }, Validators.required];
  email = ['', Validators.required];
  displayName = [''];
  photoURL = [''];
}

@Component({
  selector: 'tabu-journal-entry-form',
  templateUrl: './journal-entry-form.component.html',
  styleUrls: ['./journal-entry-form.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class JournalEntryFormComponent implements OnInit {
  _journal: IJournal | null = null;

  @Output() save = new EventEmitter<IJournal>();
  @Output() delete = new EventEmitter<IJournal>();
  @Output() copy = new EventEmitter<IJournal>();

  @Input()
  set journal(item: IJournal | null) {
    this._journal = item;
    if (item) {
      this.form.setValue({
        id: item.id,
        email: item.customer,
        displayName: item.customer,
        photoURL: item.customer
      });
    } else {
      this.form.reset();
    }
  }
  get user(): IJournal | null {
    return this._journal;
  }

  form = this.fb.group(new JournalModel());

  constructor(private fb: FormBuilder) {}

  ngOnInit() {}

  submit() {
    if (this._journal) {
      this.save.emit({
        ...this.form.value,
        uid: this._journal.id
      } as IJournal);
    }
  }

  onDelete() {}

  onCopy() {}
}
