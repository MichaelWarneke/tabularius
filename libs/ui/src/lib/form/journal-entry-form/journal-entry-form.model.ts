import { IJournal, IJournalEntry } from '@tabularius/shared/models';
import {
  FormControlTextbox,
  FormControlDate,
  Validators
} from '@tabularius/shared/dynamic-form';

export class JournalEntryFormModel implements IJournalEntry {
  account = new FormControlTextbox('', [Validators.required]);
  amount = new FormControlTextbox('', [Validators.required]);
  currency = new FormControlTextbox('AUD', [Validators.required]);
  amountForeignCurrency = new FormControlTextbox('', []);
  currencyForeign = new FormControlTextbox('', []);
  comment = new FormControlTextbox('', []);
  constructor() {
    this.account.label = 'Account';
    this.account.maxWidth = 200;
    this.amount.label = 'Amount';
    this.amount.maxWidth = 60;
    this.currency.label = 'Currency';
    this.currency.maxWidth = 60;
    this.amountForeignCurrency.label = 'Amount Foreign';
    this.amountForeignCurrency.maxWidth = 60;
    this.currencyForeign.label = 'Foreign Currency';
    this.currencyForeign.maxWidth = 60;
    this.comment.label = 'Comment';
    this.comment.maxWidth = 300;
  }
}
export class JournalFormModel implements IJournal {
  date = new FormControlDate('', [Validators.required]);
  dateOfTransaction = new FormControlDate('', []);
  dateOfEstimateTransaction = new FormControlDate('', []);
  customer = new FormControlTextbox('', []);
  supplier = new FormControlTextbox('', []);
  employee = new FormControlTextbox('', []);
  comment = new FormControlTextbox('', []);
  entries: JournalEntryFormModel[] = new Array<JournalEntryFormModel>(
    new JournalEntryFormModel()
  );
  constructor() {
    this.date.label = 'Date';
    this.dateOfTransaction.label = 'Transaction Date';
    this.dateOfEstimateTransaction.label = 'Est. Trans. Date';
    this.customer.label = 'Customer';
    this.customer.maxWidth = 150;
    this.supplier.label = 'Supplier';
    this.supplier.maxWidth = 150;
    this.employee.label = 'Employee';
    this.employee.maxWidth = 150;
    this.comment.label = 'Comment';
    this.comment.maxWidth = 300;
  }
}

export class FormSetup {
  model = new JournalFormModel();
  submitButtonText = 'Save';
  copyButtonText = 'Copy';
  deleteButtonText = 'Delete';
  resetButtonText = 'Reset';
}

export class FormEntrySetup {
  model = new JournalEntryFormModel();
  copyButtonText = 'Add';
  deleteButtonText = 'Delete';
}
