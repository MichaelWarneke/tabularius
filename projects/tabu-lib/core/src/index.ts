/*
 * Public API Surface of tabu-lib
 */

export * from './lib/core/core.module';
export * from './lib/material/material.module';


