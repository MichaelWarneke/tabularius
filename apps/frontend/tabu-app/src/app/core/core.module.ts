import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { TabuAppSidenavModule } from '@tabularius/ui';

@NgModule({
  declarations: [],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    TabuAppSidenavModule
  ],
  exports: [TabuAppSidenavModule]
})
export class CoreModule {}
