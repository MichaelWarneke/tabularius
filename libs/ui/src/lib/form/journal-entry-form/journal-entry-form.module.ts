import { NgModule } from '@angular/core';
import { JournalEntryFormComponent } from './journal-entry-form.component';
import { SharedModule } from '@tabularius/shared/ui';
import { DynamicFormModule } from '@tabularius/shared/dynamic-form';

@NgModule({
  declarations: [JournalEntryFormComponent],
  imports: [SharedModule, DynamicFormModule],
  exports: [JournalEntryFormComponent]
})
export class JournalEntryFormModule {}
