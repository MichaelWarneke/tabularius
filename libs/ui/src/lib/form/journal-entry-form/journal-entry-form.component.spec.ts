import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { JournalEntryFormComponent } from './journal-entry-form.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('JournalEntryFormComponent', () => {
  let component: JournalEntryFormComponent;
  let fixture: ComponentFixture<JournalEntryFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [JournalEntryFormComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(JournalEntryFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
