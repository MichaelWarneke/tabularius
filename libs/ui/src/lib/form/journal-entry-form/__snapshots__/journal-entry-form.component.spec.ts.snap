// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalEntryFormComponent should render 1`] = `
<tabu-journal-entry-form
  _journal={[Function Object]}
  copy={[Function EventEmitter]}
  delete={[Function EventEmitter]}
  formModel={[Function FormSetup]}
  save={[Function EventEmitter]}
>
  <h1>
    Journal Form
  </h1><tabu-dynamic-form /><hr />
  {
    "id": "123",
    "date": "2018-12-31T00:00:00.000Z",
    "dateOfTransaction": null,
    "dateOfEstimateTransaction": null,
    "supplier": null,
    "customer": "Cust",
    "employee": null,
    "comment": "Hello",
    "entries": [
      {
        "account": "office",
        "amount": 12.4,
        "currency": "AUD",
        "amountForeignCurrency": null,
        "currencyForeign": null,
        "comment": "Hello"
      }
    ]
  }
  
</tabu-journal-entry-form>
`;
